Adds CGAffineTransform

colinta opened the pull request:
CATransform3D coming soon, but this one is the smaller sibling, so tackled it first.

Version bump to 0.11.0 (should CATransform3D bump to 0.12.0 or 0.11.1?)

Also, gave `CGRect.make(point:, size:)` support for "CGPoint/CGSize-like" arrays.

colinta committed:
TravisCI

colinta committed:
tweaked CGRect.make to allow points or point-like arrays (and same for sizes)

colinta committed:
added CGAffineTransform

colinta commented:
Oh, and I guess I added TravisCI, too!  Well that wouldn't hurt.

clayallsopp commented on lib/geomotion/cg_affine_transform.rb:
```diff
@@ -0,0 +1,173 @@
+  # Concatenates the two transforms
+  # @return CGAffineTransform
+  def <<(transform)
+    CGAffineTransformConcat(self, transform)
```
Really really small thing, but why not do something like either:

``` ruby
def <<(transform
  self + transform
end
```

or  even `alias_method :<<, :+`

colinta replied:
Ah, I think I meant to _choose_ one or the other, but couldn't decide...

since -@ makes sense as 'invert', it makes sense to have +/-, but like you say < makes more sense as concat...

so both it is! :-D

I'll use alias or something there.

clayallsopp commented:
Man using `<<` makes so much more sense than chaining `CGAffineTransformConcat`, very nice :+1:

colinta committed:
CGAffineTransform changes

added concat and invert - the + << and -@ methods are aliased to the named methods
fixed some things
added to the README

colinta commented:
K, what do you think of that - added `concat` and `invert` and the operators are aliased to those methods.

colinta commented:
dang it, should run specs first ;-)

colinta committed:
hehe, ooops

clayallsopp commented:
Awesome, :shipit:

colinta commented:
haha, glad you dig - I do too!  these will be handy!

I'm working on CATransform3D now.  It's a perfect place to add some extra transforms (trapezoidal stuff) I've had around.
